mmdet3d/evaluation/functional/panoptic_seg_eval.py

```diff
@@ -0,0 +1,387 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Dict, List, Tuple
+
+import numpy as np
+from mmengine.logging import MMLogger, print_log
+
+PQReturnsType = Tuple[np.double, np.double, np.ndarray, np.ndarray, np.ndarray]
+
+
+class EvalPanoptic:
+    r"""Evaluate panoptic results for Semantickitti and NuScenes.
+    Please refer to the `semantic kitti api
+    <https://github.com/PRBonn/semantic-kitti-api/>`_ for more details
+
+    Args:
+        classes (list): Classes used in the dataset.
+        thing_classes (list): Things classes used in the dataset.
+        stuff_classes (list): Stuff classes used in the dataset.
+        min_num_points (int): Minimum number of points of an object to be
+            counted as ground truth in evaluation.
+        id_offset (int): Offset for instance ids to concat with
+            semantic labels.
+        label2cat (dict[str]): Map from label to category.
+        ignore_index (list[int]): Ignored classes in evaluation.
+        logger (logging.Logger | str, optional): Logger used for printing.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 classes: List[str],
+                 thing_classes: List[str],
+                 stuff_classes: List[str],
+                 min_num_points: int,
+                 id_offset: int,
+                 label2cat: Dict[str, str],
+                 ignore_index: List[str],
+                 logger: MMLogger = None):
+        self.classes = classes
+        self.thing_classes = thing_classes
+        self.stuff_classes = stuff_classes
+        self.ignore_index = np.array(ignore_index, dtype=int)
+        self.n_classes = len(classes)
+        self.label2cat = label2cat
+        self.logger = logger
+        self.include = np.array(
+            [n for n in range(self.n_classes) if n not in self.ignore_index],
+            dtype=int)
+        self.id_offset = id_offset
+        self.eps = 1e-15
+        self.min_num_points = min_num_points
+        self.reset()
+
+    def reset(self):
+        """Reset class variables."""
+        # general things
+        # iou stuff
+        self.px_iou_conf_matrix = np.zeros((self.n_classes, self.n_classes),
```

Comment:
what does px mean? Conf is also confusing ad it may be thought as confusion
Can we simply use `self.confusion_matrix` here.